lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -162,7 +172,7 @@ class GoogleCloudOutput < BufferedOutput
     attr_reader :running_on_managed_vm
     attr_reader :gae_backend_name
     attr_reader :gae_backend_version
-    attr_reader :service_name
+    attr_reader :resource_type
```

Comment:
I split the resource object into type and labels because the `MonitoredResource` object is different between the grpc path and non-grpc path. If `type` and `labels` are stored simply as `string` and `hash`, we can construct the two different types of `MonitoredResource` objects at the end of the method to avoid extra `if @use_grpc ... else ...` blocks along the way.